plan/logical_plan_builder.go

```diff
@@ -612,6 +612,14 @@ func (b *planBuilder) buildDistinct(child LogicalPlan, length int) LogicalPlan {
 // joinFieldType finds the type which can carry the given types.
 func joinFieldType(a, b *types.FieldType) *types.FieldType {
 	resultTp := types.NewFieldType(types.MergeFieldType(a.Tp, b.Tp))
+	// This logic need be combined with buildProjection4Union.
+	if a.Tp == mysql.TypeNewDecimal {
+		// For Decimal result be unsigned when all union children be unsigned.
```

Comment:
how about: "the decimal type could be unsigned only when all the decimals to be united are unsigned" @CaitinChen We need your help to refine this comment 😂

Reply:
@CaitinChen  need help~~

Reply:
The decimal type is unsigned only when all the decimals to be united are unsigned.

Reply:
@zz-jason @lysu I think Jason's may be right, but I don't know why you use "could be".